inc/js_scripts/admin/main.js

```diff
@@ -1,50 +1,116 @@
+			initSpectrum( {
+				target: $( '#header_color' ),
+				color: this.model.get( 'header_color' )
+			}, ( color ) => {
+				this.handleColorChange( 'header_color', color );
 			} );
+			initSpectrum( {
+				target: $( '#content_color' ),
+				color: this.model.get( 'content_color' )
+			}, ( color ) => {
+				this.handleColorChange( 'content_color', color );
+			} )
 		},
 		render: function () {
-			const _html = $( "#custom-template" ).html();
-			this.$el.html( _.template( _html )( {
+			const _html = tpl( 'views/post-content', {
 				'option': this.model.get( 'plugin_state' ),
 				'special_word': this.model.get( 'special_word' ),
+			} );
+			this.$el.html( _html );
+		},
+		events: {
+			'change .state, #special-word': 'handleChange',
+		},
+		handleChange: function ( event ) {
+			this.model.set( $( event.target ).attr( 'data-model_attr_name' ), $( event.target ).val() );
+		},
+		handleColorChange: function ( attr, color ) {
+			this.model.set( attr, color );
+		}
+	} )
+
+	let PostMetadataView = Backbone.View.extend( {
+		initialize: function () {
+			this.render();
+		},
+		render: function () {
+			const _html = tpl( 'views/post-metadata', {
 				'date_format': this.model.get( 'custom_date_format' )
-			} ) );
+			} );
+			this.$el.html( _html );
 		},
 		events: {
-			'change .state, #special-word, #custom_date_format': 'handleChange',
-			'change .color-picker': 'handleColorChange',
-			'click #save-settings': 'saveSettings',
+			'change #custom_date_format': 'handleChange',
 		},
 		handleChange: function ( event ) {
 			this.model.set( $( event.target ).attr( 'data-model_attr_name' ), $( event.target ).val() );
+		}
+	} )
+
+	let SettingsView = Backbone.View.extend( {
+		initialize: function () {
+			this.render();
+		},
+		render: function () {
+			this.$el.append( this.attributes.content.$el );
+			this.$el.append( this.attributes.metadata.$el );
+			const _html = tpl( 'views/main' );
+			this.$el.append( _html );
 		},
-		handleColorChange: function ( event ) {
-			this.model.set( $( event.target ).attr( 'data-model_attr_name' ), $( event.target ).spectrum( "get" ).toHex() );
-			console.log( this.model.get( 'header_color' ) );
-			console.log( this.model.get( 'content_color' ) );
+		events: {
+			'click #save-settings': 'saveSettings',
 		},
 		saveSettings: function () {
-			$.ajax( {
-				method: 'POST',
-				url: post_modifier.rest_url,
-				data: this.model.attributes,
-			} ).success( function () {
-				alert( "Settings saved" );
+			model.save();
+		}
+	} );
+
+	let PreviewView = Backbone.View.extend( {
+		initialize: function () {
+
+			const _html = tpl( 'views/live-preview', {
+				post_date: post_modifier.preview.rand_post_date,
+				post_title: post_modifier.preview.rand_post_title,
+				post_content: post_modifier.preview.rand_post_content
 			} );
+			this.$el.find( '.actual-preview' ).append( _html );
+
+			this.render();
+			this.model.on( 'change:header_color', () => {
+				this.render();
+			} )
```

Comment:
if it fits on one line, you can do this: ( and skip the es6 `() => { }` )
```suggestion
			this.model.on( 'change:header_color', this.render )
```
same thing for 'change:content_color